packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,187 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { warnOnce } from "../utils/warn-once";
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & { quality?: number };
+
+/**
+ * max(...imageSizes) must be less then min(...deviceSizes)
+ * idea is to support images with vw size less than 100vw
+ * https://github.com/vercel/next.js/blob/canary/packages/next/shared/lib/image-config.ts
+ **/
+const imageSizes = [16, 32, 48, 64, 96, 128, 256, 384];
+const deviceSizes = [640, 750, 828, 1080, 1200, 1920, 2048, 3840];
+
+const allSizes = [...imageSizes, ...deviceSizes];
+
+/**
+ * https://developers.cloudflare.com/images/image-resizing/url-format/
+ **/
+const cloudflareImageLoader = ({
+  width,
+
+  src,
+  quality,
+}: {
+  width: number;
+  src: string;
+  quality?: number;
+}) => {
+  let resizeOrigin = "";
+
+  if (process.env.NODE_ENV !== "production") {
+    if (!allSizes.includes(width)) {
+      warnOnce("Width must be only from allowed values");
+    }
+
+    // No image transformation in development
+    resizeOrigin = "https://webstudio.is";
+  }
+
+  const params = [
+    `width=${width}`,
+    quality != null && `quality=${quality}`,
+    // TODO: Must be checked how it works with caching (like avif image is cached and then read by the browser without avif support)
+    "format=auto",
+  ].filter(Boolean);
```

Comment:
there is a API for this: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
Cloudflare especially wrote in docs:
The path is not URL-encoded, so the resizing URL can be safely constructed by concatenating /cdn-cgi/image/options and the original image URL. So its just path without need of any url encoding. URLSearchParams has nothing with path creating, URL is also not needed here

Reply:
I see, in this case, its not a correctness thing, more like a style thing. Reading it I had no idea if its gonna be encoded later, so I would much prefer to use a safe API. Its just better to standardize on using new URL and new URLSearchParams IMO

Reply:
Next person in the code base will read your code and build a URL like you did, but in a place where its unsafe. Just don't give them a reason to copy this.

Reply:
Now no url.